terraform/routines.tf

```diff
@@ -173,3 +173,43 @@ resource "google_bigquery_routine" "user_agent_parser" {
   }
 EOF
 }
+
+resource "google_bigquery_routine" "parse_furnished_types" {
+  dataset_id   = local.routines_dataset
+  routine_id   = "parse_furnished_types${local.branch_suffix_underscore_edition}"
+  routine_type = "SCALAR_FUNCTION"
+  language     = "JAVASCRIPT"
+  arguments {
+    name          = "raw_types"
+    argument_kind = "FIXED_TYPE"
+    data_type     = "{\"typeKind\": \"ARRAY\", \"arrayElementType\": {\"typeKind\": \"STRING\"}}"
+  }
+  definition_body = <<EOF
+if (raw_types == null || raw_types.length == 0) {
+  return null;
+}
+
+final_types = [];
+raw_types = raw_types.map(raw_type => raw_type.toLowerCase());
+
+if (raw_types.filter(t => t.includes("shell") || t.includes("kaal"))) {
```

Comment:
We should discuss if we need to implement more logic here to reduce the array of values to a single one. In practice we never really use the array and this just means we will have to implement the same logic in multiple places downstream from here.

Reply:
not sure what you mean here, perhaps do a call